### ISSUE TITLE
Some ideas

### ISSUE THREAD

**Issue by SkyKrye** [megaboi2005/bark-it#10]
So basically I was thinking (emphasis on thinking) that if maybe bark-it would start a hiring system for more devs this could've been a better and faster process.
Basically some ideas:
- make a post managing system
- authors of posts have accessibility to commenters' comments and can heart, pin, or report them
- a working nsfw filter to just delete posts or have words in the posts have a bleeped out black bar on them
- *if you really really want to* then have a seperate option for making nsfw posts which will require the user's account to be have over 18 years of set age, which the post will be highlighted in red
- for nsfw images just have some code which blurs out red-peach stuff in an image
- have a tagging system; basically accounts that can be tagged with @exampleuser
- have a trending posts system; basically posts within the last 24 hours that get more than a specific amount of likes
- accounts with special accesses; much like discord an account can have roles, for example someone can have the "admin" role and "contributer" role
- ideas system; people can add posts to tell them some ideas, this depends on you guys if you wanna stick with guthub for this or not
- finally a status message; where accounts can add a status which sticks to their profile as a special post (25 characters max)

thats all ig, gonna be a hard time to code but still just some proposals if you agree with them

**Comment by megaboi2005** [comment]
no nsfw, thats gonna be one unique thing about this site. 
The idea of a post managing system is vague, what do you mean by that?
Tagging system is impossible as of now since we don't have a notifications system.
Trending posts might be coming after we finish the layout.
Accounts with different "roles" would just cause a big hierarchy mess.
Statuses would be cool but on other social medias they call them bios (idk why tbh).

**Comment by Bigjango13** [comment]
I was thinking that most (if not all) of the biggest social medias have a quirk, Reddit has subreddits, Twitter has tweets, etc... I was thinking about what bark-it should be and I had an idea; channels. I don't mean channels as accounts but channels as in feeds (channels of information). Since we have already decided not to have an AI generated feed we will need a way to let the users decide what they see. My idea is a user can subscribe to a users posts, a topics posts, or another users channel. When a new posts is added in a topic it is inputted into the feed of all users that are subscribed to it (same thing when a user posts). This makes up a users channel (aka their feed). If you like what another user has subscribed to you can subscribe to their feed and have a small percent of it added to your channel.

**Comment by megaboi2005** [comment]
> I was thinking that most (if not all) of the biggest social medias have a quirk, Reddit has subreddits, Twitter has tweets, etc... I was thinking about what bark-it should be and I had an idea; channels. I don't mean channels as accounts but channels as in feeds (channels of information). Since we have already decided not to have an AI generated feed we will need a way to let the users decide what they see. My idea is a user can subscribe to a users posts, a topics posts, or another users channel. When a new posts is added in a topic it is inputted into the feed of all users that are subscribed to it (same thing when a user posts). This makes up a users channel (aka their feed). If you like what another user has subscribed to you can subscribe to their feed and have a small percent of it added to your channel. 

I was thinking of something exactly like that. It's a really good idea and I also think trending channels should also be a thing as well. No ai, just the amount of clicks and such.

**Comment by Bigjango13** [comment]
We will have some challenges, such as if someone you are following the channel of is following your channel it could lead to unbreakable loops. I think this could be overcome by having a list of people a post has already been put in the channel of.

**Comment by SkyKrye** [comment]
> 1. Do you want a separate indication, like a checkbox, if a user wishes to view a percentage of the other users' feed, or do you want it as an automatic?
> 2. What should be shown as the default feed for someone who just created an account and hasn't subscribed to anything? If AI generated feeds aren't going to be used, does that include gathering info on the most popular feeds and putting it up?
> 3. What model are we going for? Is Bark-It going to be focusing on just Twitter aspects? Is it going to be mainly similar to tweets, and focus on text? Or do we wish to push more on videos? Also, I think audio could be something that we could implement. Like, no video. Just text to speech or audio. Because I know a lot of friends who instead of typing in Instagram send messages in an audio format as they don't want to type it out.

in my opinion, a user who just joins barkit or doesnt make an account and views logged out (although that's bad for privacy) they can view the top posts, this is kind of a user gaining strategy as newer people will be inspired and maybe create accounts, the top posts can obviously be sorted as well with newer posts _with_ them, if we were to go with the old style of barkit to show every post made then it would cause a heck lot of problems like lag etc, so it'd be better to just have it equalized for top posts and posts with less recognition

**Comment by SkyKrye** [comment]
barkit shouldn't be a more twitter based platform but more like reddit and twitter along with the videos and audios feature of youtube, this'll keep some stuff equalized and balanced and there could be a heck ton of text sh*tposters who would be needed to filtered out as they're simply irrelevant or just spamming

**Comment by SkyKrye** [comment]
> I was thinking that most (if not all) of the biggest social medias have a quirk, Reddit has subreddits, Twitter has tweets, etc... I was thinking about what bark-it should be and I had an idea; channels. I don't mean channels as accounts but channels as in feeds (channels of information). Since we have already decided not to have an AI generated feed we will need a way to let the users decide what they see. My idea is a user can subscribe to a users posts, a topics posts, or another users channel. When a new posts is added in a topic it is inputted into the feed of all users that are subscribed to it (same thing when a user posts). This makes up a users channel (aka their feed). If you like what another user has subscribed to you can subscribe to their feed and have a small percent of it added to your channel.

yes, and the channels system could be targeted at a user using only what areas they visit and not some data collection thing like facebook or whatnot, however that system **can** be also used like discord, where a big community has multiple sub-communities as text or hangout areas, similarly to reddit flairs sorting out posts there can just be entire sections which imo makes stuff kinda neater and less problematic for a proper community system, anyway I say all the best for coding in these  obscure ideas, and good luck!
